### PR TITLE
ConnectionPolicy: Refactors Code to Reduce Default Request Timeout to 6 Seconds.

### LINKED ISSUE #3895: Request Timeout - Reduce Default Timeout to 5 Seconds
This goal of this task to reduce the default request timeout to `5` seconds, from `10` seconds.

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

- **Reduce Default Timeout to 6 Seconds:** This change primarily refactors the `ConnectionPolicy` class in the v3 SDK to reduce the default request timeout to `6` seconds from `10` seconds.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: closes #3895

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ConnectionPolicy.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.Azure.Cosmos
     /// </summary>
     internal sealed class ConnectionPolicy
     {
-        private const int defaultRequestTimeout = 10;
+        private const int defaultRequestTimeout = 6;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/ConnectionPolicy.cs:22]: Please update CosmosClientOptions.RequestTimeout API documentation

**Review by FabianMeiswinkel — DISMISSED** [review]
LGTM except for the API doc update

File: Microsoft.Azure.Cosmos/src/ConnectionPolicy.cs (hunk 1 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/ConnectionPolicy.cs:22]: Updated.

**Comment by ealsur** [comment]
@msftbot wait for @kirankumarkolli to approve just to confirm we are all on the same page

**Review by ealsur — APPROVED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**